### PR TITLE
[CARBONDATA-1891] Fixed timeseries table creation after data load

### PR DESCRIPTION
**Pull request by kunal642**
**Analysis:** CreatePreAggregateTableCommand.run() was being called from CarbonCreateDataMapCommand.processMetaData() Due to this wrong table was being created which is not in DataMapSchemaList.

**Solution:** Refactor code to call CreatePreAggregateTableCommand.processMetadata instead of run.

Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [x] Any interfaces changed?
 No
 - [x] Any backward compatibility impacted?
 No
 - [x] Document update required?
No
 - [x] Testing done
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
        - How it is tested? Please attach test report.
        - Is it a performance related change? Please attach the performance test report.
        - Any additional information to help reviewers in testing this change.
       
 - [x] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA. 
No

### CODE REVIEW

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/preaaggregate/PreAggregateUtil.scala [hunk 1]
```diff
@@ -514,9 +514,16 @@ object PreAggregateUtil {
...
       parentCarbonTable.getTableName, validateSegments.toString)
-    val headers = parentCarbonTable.getTableInfo.getDataMapSchemaList.asScala.
-      find(_.getChildSchema.getTableName.equals(dataMapIdentifier.table)).get.getChildSchema.
-      getListOfColumns.asScala.map(_.getColumnName).mkString(",")
+    val dataMapSchemas = parentCarbonTable.getTableInfo.getDataMapSchemaList.asScala
+    val headers = dataMapSchemas.find(_.getChildSchema.getTableName.equalsIgnoreCase(
+      dataMapIdentifier.table)) match {
+      case Some(dataMapSchema) =>
+        dataMapSchema.getChildSchema.getListOfColumns.asScala.map(_.getColumnName).mkString(",")
+      case None =>
```
Comment (kumarvishal09) [inline integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/preaaggregate/PreAggregateUtil.scala]: when this condition will come??

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/693/

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/preaaggregate/PreAggregateUtil.scala (hunk 1 above)
Comment (kunal642) [reply integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/preaaggregate/PreAggregateUtil.scala]: This is just to throw a proper exception if ever this condition is met. The message will tell which aggregate table is not present in the datamap schema list.

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1922/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/2253/

**Comment by kumarvishal09** [comment]
LGTM

**Pull request closed without merging**